tests/functional/aws-node-sdk/test/object/objectCopy.js

```diff
@@ -0,0 +1,834 @@
+    overwriteme: 'wiped',
+};
+const content = 'I am the best content ever';
+
+const otherAccountBucketUtility = new BucketUtility('lisa', {});
+const otherAccountS3 = otherAccountBucketUtility.s3;
+
+function checkNoError(err) {
+    assert.equal(err, null,
+        `Expected success, got error ${JSON.stringify(err)}`);
+}
+
+function checkError(err, code) {
+    assert.notEqual(err, null, 'Expected failure but got success');
+    assert.strictEqual(err.code, code);
+}
+
+function dateFromNow(diff) {
+    const d = new Date();
+    d.setHours(d.getHours() + diff);
+    return d.toISOString();
+}
+
+function dateConvert(d) {
+    return (new Date(d)).toISOString();
+}
+
+describe('Object Copy', () => {
+    withV4(sigCfg => {
+        let bucketUtil;
+        let s3;
+        let etag;
+        let etagTrim;
+        let lastModified;
+
+        before(() => {
+            bucketUtil = new BucketUtility('default', sigCfg);
+            s3 = bucketUtil.s3;
+            return bucketUtil.empty(sourceBucketName)
+            .then(() =>
+                bucketUtil.empty(destBucketName)
+            )
+            .then(() =>
+                bucketUtil.deleteMany([sourceBucketName, destBucketName])
+            )
+            .catch(err => {
+                if (err.code !== 'NoSuchBucket') {
+                    process.stdout.write(`${err}\n`);
+                    throw err;
+                }
+            })
+            .then(() => bucketUtil.createOne(sourceBucketName)
+            )
+            .then(() => bucketUtil.createOne(destBucketName)
+            )
+            .catch(err => {
+                throw err;
+            });
+        });
+
+        beforeEach(() => s3.putObjectAsync({
+            Bucket: sourceBucketName,
+            Key: sourceObjName,
+            Body: content,
+            Metadata: originalMetadata,
+        }).then(res => {
+            etag = res.ETag;
+            etagTrim = etag.substring(1, etag.length - 1);
+            return s3.headObjectAsync({
+                Bucket: sourceBucketName,
+                Key: sourceObjName,
+            });
+        }).then(res => {
+            lastModified = res.LastModified;
+        }));
+
+        afterEach(() => bucketUtil.empty(sourceBucketName)
+            .then(() => bucketUtil.empty(destBucketName))
+        );
+
+        after(() => bucketUtil.deleteMany([sourceBucketName, destBucketName]));
+
+        function requestCopy(fields, cb) {
+            s3.copyObject(Object.assign({
+                Bucket: destBucketName,
+                Key: destObjName,
+                CopySource: `${sourceBucketName}/${sourceObjName}`,
+            }, fields), cb);
+        }
+
+        it('should copy an object from a source bucket to a different ' +
+            'destination bucket and copy the metadata if no metadata directve' +
+            'header provided', done => {
+            s3.copyObject({ Bucket: destBucketName, Key: destObjName,
```

Comment:
Since most of the tests are similar, this could be consolidated into a util function and used throughout the test suite.